File: playground/c/average_grade.c
#include <stdio.h>

int main()
{
    char name[20];
    int subjects, _;
    float total_grade = 0.0;

    printf("Enter student name: ");
    scanf("%20s", &name);
    printf("\nHow many subjects do you have? > ");
    scanf("%i", &subjects);

    float grades[subjects];

    for (_ = 0; _ < subjects; _++)
    {
        printf("\nEnter grade #%i: ", _ + 1);
        scanf("%f", &grades[_]);
    }
    putchar('\n');
    for (_ = 0; _ < subjects; _++)
    {
        printf("Subject #%i: %f\n", _ + 1, grades[_]);
        total_grade += grades[_];
    }
    total_grade /= subjects;
    printf("Total: %f\n", total_grade);
    return 0;
}

File: playground/java/AverageGrade.java
import java.util.Scanner;

public class AverageGrade
{
    public static void main(String[] args)
    {
        // ? Initialize and declare variables.
        var user_input = new Scanner(System.in);
        String name;
        int number_of_subjects;
        float average = 0.0F;  // Add default value for average grade.

        // ? Ask for user input.
        System.out.print("Enter the student's name: ");
        name = user_input.nextLine();
        System.out.print("How many subjects does the student have? > ");
        number_of_subjects = user_input.nextInt();

        // ? Create a new float array with length of <number_of_subjects>.
        float[] subject_grades = new float[number_of_subjects];

        // ? Initiate for loop to fill up the array with student's grades.
        for (int i = 0; i < number_of_subjects; i++)  // This is called a "for" loop.
        {
            System.out.printf("Enter the student's grade for subject #%s: ", i + 1);
            subject_grades[i] = user_input.nextFloat();
        }

        // ? Print the output.
        System.out.printf("\n%s's grades:\n", name);
        for (var subject_grade : subject_grades)  // This is called a "for each" loop.
        {
            System.out.printf("- %s%s\n", subject_grade, '%');
            average += subject_grade;  // While printing the grades, add it to <average>.
        }
        average = average / number_of_subjects;  // Divide the current value of <average> to get the actual average.
        System.out.printf("\nAverage Grade: %s%s", average, '%');
    }
}
